### PR TITLE
boards: arm: bmd_345_eval: Add BMD 345 EVAL Board Support

### LINKED ISSUE #31585: BMD345: Extended BLE range with PA/LNA
I am using [BMD-345 Eval kit](https://www.mouser.com/datasheet/2/1025/BMD-345_DataSheet_(UBX-19039908)-1879743.pdf) from ubox  to extended the BLE range using PA/LNA.

How to enable PA/LNA feature in zephyr to achieve long range using dynamic method?

-> **BT_CTLR_GPIO_LNA_PIN** Kconfig is used to control LNA pin.
-> **BT_CTLR_GPIO_PA_PIN** Kconfig is used to control PA pin

what about **MODE** and **A-SEL** pin?

### PR DESCRIPTION
**Pull request by Navin-Sankar**
Add support for BMD 345 EVAL Board definition with PA/LNA support

fixes: #31585

Signed-off-by: NavinSankar Velliangiri <navin@linumiz.com>

### CODE REVIEW

File: boards/arm/bmd_345_eval/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+* BUTTON4 = SW4 = P0.25
+* BOOT = SW5 = boot/reset
+
+Programming and Debugging
+*************************
+
+Applications for the ``bmd_345_eval`` board configuration can be
+built and flashed in the usual way (see :ref:`build_an_application`
+and :ref:`application_run` for more details); however, the standard
+debugging targets are not currently available.
```
Comment (MaureenHelm) [inline boards/arm/bmd_345_eval/doc/index.rst:103]: Is that true, given you've added jlink runner support in `boards/arm/bmd_345_eval/board.cmake`?

Comment (Navin-Sankar) [reply boards/arm/bmd_345_eval/doc/index.rst]: Sorry for the mistake, it support's jlink runner support.

**Review by MaureenHelm — APPROVED** [review]

File: CODEOWNERS [hunk 2]
```diff
@@ -79,6 +79,7 @@
 /boards/arm/96b_stm32_sensor_mez/         @Mani-Sadhasivam
 /boards/arm/96b_wistrio/                  @Mani-Sadhasivam
 /boards/arm/arduino_due/                  @ioannisg
+/boards/arm/bmd_345_eval/		  @Navin-Sankar
```
Comment (cvinayak) [inline CODEOWNERS:82]: Replace tabs with spaces for alignments

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Review by ioannisg — APPROVED** [review]

**Comment by MaureenHelm** [comment]
@cvinayak please revisit

**Review by cvinayak — APPROVED** [review]

**Pull request merged**